src/substrait/textplan/SymbolTablePrinter.cpp

```diff
@@ -432,6 +435,199 @@ std::string SymbolTablePrinter::outputToText(const SymbolTable& symbolTable) {
   return text.str();
 }
 
+void SymbolTablePrinter::addInputsToRelation(
+    const SymbolInfo& symbolInfo,
+    ::substrait::proto::Rel* relation) {
+  auto relationData = ANY_CAST(std::shared_ptr<RelationData>, symbolInfo.blob);
+
+  // Connect up the incoming inputs in the relation data to the appropriate
+  // input/left/right/inputs of this relation (which recursively also needs to
+  // have its inputs added).
+  if (symbolInfo.subtype.type() != typeid(RelationType)) {
+    // The type isn't one we expected bail.
+    return;
+  }
+  auto relationType = ANY_CAST(RelationType, symbolInfo.subtype);
+  switch (relationType) {
+    case RelationType::kRead:
+      // No inputs to add.
+      break;
+    case RelationType::kProject:
+      if (relationData->continuingPipeline != nullptr) {
+        auto continuingRelationData = ANY_CAST(
+            std::shared_ptr<RelationData>,
+            relationData->continuingPipeline->blob);
+        *relation->mutable_project()->mutable_input() =
+            continuingRelationData->relation;
+        addInputsToRelation(
+            *relationData->continuingPipeline,
+            relation->mutable_project()->mutable_input());
+      }
+      break;
+    case RelationType::kJoin:
+      if (relationData->newPipelines.size() == kBinaryRelationInputCount) {
+        auto leftRelationData = ANY_CAST(
+            std::shared_ptr<RelationData>, relationData->newPipelines[0]->blob);
```

Comment:
So this means...

```
read1 -> project1 -> join1
read2 -> project2 -> join1
```

In this case `project1` will be the `left` and `project2` will be the `right`?